frontend/src/components/panels/Document.vue

```diff
@@ -1,70 +1,72 @@
 <template>
 	<LayoutCol :class="'document'">
 		<LayoutRow :class="'options-bar'">
-			<div class="left side">
-				<DropdownInput :menuEntries="documentModeEntries" v-model:selectedIndex="documentModeSelectionIndex" :drawIcon="true" />
-
-				<Separator :type="SeparatorType.Section" />
-
-				<ToolOptions :activeTool="activeTool" />
-			</div>
-			<div class="spacer"></div>
-			<div class="right side">
-				<OptionalInput v-model:checked="snappingEnabled" @update:checked="comingSoon(200)" :icon="'Snapping'" title="Snapping" />
-				<PopoverButton>
-					<h3>Snapping</h3>
-					<p>The contents of this popover menu are coming soon</p>
-				</PopoverButton>
-
-				<Separator :type="SeparatorType.Unrelated" />
-
-				<OptionalInput v-model:checked="gridEnabled" @update:checked="comingSoon(318)" :icon="'Grid'" title="Grid" />
-				<PopoverButton>
-					<h3>Grid</h3>
-					<p>The contents of this popover menu are coming soon</p>
-				</PopoverButton>
-
-				<Separator :type="SeparatorType.Unrelated" />
-
-				<OptionalInput v-model:checked="overlaysEnabled" @update:checked="comingSoon(99)" :icon="'Overlays'" title="Overlays" />
-				<PopoverButton>
-					<h3>Overlays</h3>
-					<p>The contents of this popover menu are coming soon</p>
-				</PopoverButton>
-
-				<Separator :type="SeparatorType.Unrelated" />
-
-				<RadioInput :entries="viewModeEntries" v-model:selectedIndex="viewModeIndex" />
-				<PopoverButton>
-					<h3>View Mode</h3>
-					<p>The contents of this popover menu are coming soon</p>
-				</PopoverButton>
-
-				<Separator :type="SeparatorType.Section" />
-
-				<NumberInput @update:value="setRotation" v-model:value="documentRotation" :incrementFactor="15" :unit="`°`" ref="rotation" />
-
-				<Separator :type="SeparatorType.Section" />
-
-				<IconButton :action="() => this.$refs.zoom.onIncrement(IncrementDirection.Increase)" :icon="'ZoomIn'" :size="24" title="Zoom In" />
-				<IconButton :action="() => this.$refs.zoom.onIncrement(IncrementDirection.Decrease)" :icon="'ZoomOut'" :size="24" title="Zoom Out" />
-				<IconButton :action="() => this.$refs.zoom.updateValue(100)" :icon="'ZoomReset'" :size="24" title="Zoom to 100%" />
-
-				<Separator :type="SeparatorType.Related" />
-
-				<NumberInput
-					v-model:value="documentZoom"
-					@update:value="setCanvasZoom"
-					:min="0.000001"
-					:max="1000000"
-					:incrementBehavior="IncrementBehavior.Callback"
-					:incrementCallbackIncrease="increaseCanvasZoom"
-					:incrementCallbackDecrease="decreaseCanvasZoom"
-					:unit="`%`"
-					:displayDecimalPlaces="4"
-					ref="zoom"
-				/>
-			</div>
+			<FloatingMenuToggleGroup>
```

Comment:
I haven't heard the word 'spawner' in a UI context before but I suppose it would make sense. Although to be pedantic, it does also 'unspawn'. You could even say it 'toggles' between spawned and unspawned states :) Toggle is admittedly a bit of an overloaded term. Here I mean it as a verb, but yeah it can also be a noun referring to a specific style of button.

What about simplifying the naming all the way down to something like "MenuButton" and "MenuButtonGroup"?

Reply:
Or maybe "MenuTriggerButton"

Reply:
I'd be inclined to go with `FloatingMenuGroup` since dropdowns, popover buttons, etc. should all fit within a group.